libcxx/include/__expected/expected.h

```diff
@@ -920,9 +920,9 @@ class expected : private __expected_base<_Tp, _Err> {
     requires is_constructible_v<_Err, _Err&>
   _LIBCPP_HIDE_FROM_ABI constexpr auto and_then(_Func&& __f) & {
     using _Up = remove_cvref_t<invoke_result_t<_Func, _Tp&>>;
-    static_assert(__is_std_expected<_Up>::value, "The result of f(**this) must be a specialization of std::expected");
+    static_assert(__is_std_expected<_Up>::value, "The result of f(value()) must be a specialization of std::expected");
```

Comment:
This may be a bit misleading as `value()` can be ill-formed (because the error type is not copyable) while monadic operations being well-formed. See also [LWG3938](https://cplusplus.github.io/LWG/issue3938) and [LWG3940](https://cplusplus.github.io/LWG/issue3940).

The resolution of [LWG3973](https://cplusplus.github.io/LWG/issue3973) uses _`val`_ (exposition-only member name). Perhaps we can use `val` or `__val()` (libc++-specific function name) in the error messages instead.

Reply:
Yeah, I think so. Since https://reviews.llvm.org/D154116 we have addressed LWG3973's std::expected part. This PR improves static_assert message. `__val` seems more closely to the LWG wording. Using `__val()` may requires the user to understand implementation details(e.g. __expected_base). @philnik777 Do you have any suggestions for this?

Reply:
Maybe we could just say something like "The result type of the functor must be a specialization of std::expected"? Most people will probably just have a lambda for it anyways. @cjdb might also have thoughts on how to word this.